code/game/objects/items/weapons/twohanded.dm

```diff
@@ -373,7 +374,8 @@
 		if(!isliving(target))
 			return
 		else
-			target.Weaken(4)
+			if(isrealchainsaw == 1)

```

Comment:
I almost think it would be better to do an istype check here over a new variable.